exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/DrillOperatorTable.java

```diff
@@ -26,34 +33,88 @@
 import org.apache.calcite.sql.SqlOperatorTable;
 import org.apache.calcite.sql.SqlSyntax;
 import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+import org.apache.drill.exec.planner.physical.PlannerSettings;
+import org.apache.drill.exec.server.options.SystemOptionManager;
 
 import java.util.List;
+import java.util.Map;
 
+/**
+ * Implementation of {@link SqlOperatorTable} that contains standard operators and functions provided through
+ * {@link #inner SqlStdOperatorTable}, and Drill User Defined Functions.
+ */
 public class DrillOperatorTable extends SqlStdOperatorTable {
-  static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(DrillOperatorTable.class);
-
+//  private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(DrillOperatorTable.class);
   private static final SqlOperatorTable inner = SqlStdOperatorTable.instance();
-  private List<SqlOperator> operators;
-  private ArrayListMultimap<String, SqlOperator> opMap = ArrayListMultimap.create();
+  private final List<SqlOperator> operatorsCalcite = Lists.newArrayList();
+  private final List<SqlOperator> operatorsDefault = Lists.newArrayList();
+  private final List<SqlOperator> operatorsInferernce = Lists.newArrayList();
+  private final Map<SqlOperator, SqlOperator> calciteToWrapper = Maps.newIdentityHashMap();
+
+  private final ArrayListMultimap<String, SqlOperator> opMapDefault = ArrayListMultimap.create();
+  private final ArrayListMultimap<String, SqlOperator> opMapInferernce = ArrayListMultimap.create();
+
+  private final SystemOptionManager systemOptionManager;
 
   public DrillOperatorTable(FunctionImplementationRegistry registry) {
-    operators = Lists.newArrayList();
-    operators.addAll(inner.getOperatorList());
+    this(registry, null);
+  }
 
+  public DrillOperatorTable(FunctionImplementationRegistry registry, SystemOptionManager systemOptionManager) {
     registry.register(this);
+    operatorsCalcite.addAll(inner.getOperatorList());
+    populateWrappedCalciteOperators();
+    this.systemOptionManager = systemOptionManager;
+  }
+
+  public void addDefault(String name, SqlOperator op) {
```

Comment:
From names, the functionality of addDefault and addInference is not obvious.

Please add docs to these methods and class, since implementations of pluggable function registry have to use these methods.